### DISCUSSION TITLE
Add visualization as python script & R script Like Powerbi

### DISCUSSION THREAD

**Discussion by saberhosseini** [apache/superset#20233] [Ideas]
Adding Python script or R Script as Visual is very necessary.
Is there a solution?

**Comment by Synarcs** [comment]
I think that is possible by creating a custom viz plugin which inherits the default table plugin, only thing as in PowerBi where python script can be imported for further filtering and dataset loading, in superset you could call the build query to connect it to an internal FAB endpoints, which can execute the script and return the payload of filter, which you custom plugin could just render out

**Reply by Synarcs** [reply]
This would really be a great feature to implement any more implementation ideas on this ?

**Comment by saberhosseini** [comment]
I mean, the existence of this feature in SUPERSET.
![tempsnip](https://user-images.githubusercontent.com/61079726/185052835-3f16affb-9bf6-487a-a42b-15ca8d9051d0.jpg)

**Comment by asmaier** [comment]
I fully agree with the proposal here. I think it is kind of ironic, that Apache Superset is written in Python, but does not allow to execute Python Scripts, like e.g. PowerBI does. It would be really nice to have similar possibilities as PowerBI, see e.g. https://pycaret.gitbook.io/docs/learn-pycaret/official-blog/machine-learning-in-power-bi-using-pycaret

**Comment by mike-fischer-ml** [comment]
this would be quite helpful
